src/plugins/data/public/ui/search_bar/search_bar.tsx

```diff
@@ -105,59 +108,14 @@ class SearchBarUI extends Component<SearchBarProps, State> {
     showFilterBar: true,
     showDatePicker: true,
     showAutoRefreshOnly: false,
+    filterBarPopoverHasOwnFocus: true,
   };
 
   private services = this.props.kibana.services;
   private savedQueryService = this.services.data.query.savedQueries;
   public filterBarRef: Element | null = null;
   public filterBarWrapperRef: Element | null = null;
 
-  public static getDerivedStateFromProps(nextProps: SearchBarProps, prevState: State) {
```

Comment:
This was some very convoluted for updating query from props. I think a simpler version can be done in `componentDidUpdate`.